syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package proto_transport_catalogue;

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message Stop {
	string name = 1;
	Coordinates coordinates = 2;
}

message Bus {
	string name = 1;
	repeated string route = 2;
	bool ring_route = 3;
}

message Distance {
	string from = 1;
	string to = 2;
	int32 distance = 3;
}

message TransportCatalogue {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Distance distances = 3;
	
	proto_map_renderer.RenderSettings render_settings = 4;
	proto_transport_router.TransportRouter router = 5;
}